cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -2465,4 +2465,22 @@ const char* elt_int32_utf8_utf8_utf8_utf8_utf8(
       return nullptr;
   }
 }
+
+FORCE_INLINE
+int32_t instr_utf8(const char* string, int32_t string_len, const char* substring,
+                   int32_t substring_len) {
+  if (substring_len == 0) {
+    return 1;
+  }
+
+  int32_t end_idx =
```

Comment:
if (string_len < substring_len) - shouldn't this return 0 without any further checking?

Reply:
else, your end_idx = string_len - substring_len;

And, your for loop should go from i = 0; to i <= end_idx

Reply:
Fixed